src/prommis/solvent_extraction/solvent_extraction.py

```diff
@@ -131,28 +125,103 @@ def build(self):
+                            == -b.mscontactor.aqueous[
+                                t, b.mscontactor.elements.next(s)
+                            ].get_material_flow_terms(aqueous.phase_list, m)
+                            * b.partition_coefficient[s,(k, l, m)]
+                        )
             else:
-                return (
-                    b.mscontactor.material_transfer_term[t, s, (k, l, m)]
-                    == -b.mscontactor.aqueous[
-                        t, b.mscontactor.elements.prev(s)
-                    ].get_material_flow_terms(phase_list, m)
-                    * b.partition_coefficient[(k, l, m)]
-                )
+                organic = b.mscontactor.organic
+                if self.config.organic_stream.flow_direction == FlowDirection.forward:
+                    if s == b.mscontactor.elements.first():
+                        return (
+                            b.mscontactor.material_transfer_term[t, s, (k, l, m)]
+                            == b.mscontactor.organic_inlet_state[t].get_material_flow_terms(
+                                organic.phase_list, m
+                            )
+                            * b.partition_coefficient[s,(k, l, m)]
+                        )
+                    else:
+                        return (
+                            b.mscontactor.material_transfer_term[t, s, (k, l, m)]
+                            == b.mscontactor.organic[
+                                t, b.mscontactor.elements.prev(s)
+                            ].get_material_flow_terms(organic.phase_list, m)
+                            * b.partition_coefficient[s,(k, l, m)]
+                        )
+                else:
+                    if s == b.mscontactor.elements.last():
+                        return (
+                            b.mscontactor.material_transfer_term[t, s, (k, l, m)]
+                            == b.mscontactor.organic_inlet_state[t].get_material_flow_terms(
+                                organic.phase_list, m
+                            )
+                            * b.partition_coefficient[s,(k, l, m)]
+                        )
+                    else:
+                        return (
+                            b.mscontactor.material_transfer_term[t, s, (k, l, m)]
+                            == b.mscontactor.organic[
+                                t, b.mscontactor.elements.next(s)
+                            ].get_material_flow_terms(organic.phase_list, m)
+                            * b.partition_coefficient[s,(k, l, m)]
+                        )
+
 
         self.mass_transfer_constraint = Constraint(
             self.flowsheet().time,
             self.mscontactor.elements,
             self.mscontactor.stream_component_interactions,
             rule=mass_transfer_term,
         )
+
+        self.clean_sx_pe_tank_cap = Var(
+            initialize=1405, units=units.gal, doc="Volume of polyethylene tanks"
+        )
```

Comment:
As context, these are the SX parameters used for costing. I'm not sure if these should be Vars or mutable Params